### ISSUE TITLE
Incorrect error message for resource collision

### ISSUE THREAD

**Issue by lindhe** [kubernetes-sigs/kustomize#5246]
### What happened?

I added two YAML files that had a resource collision. This correctly gave me an error saying that I "may not add resource with an already registered id: …" but it also, in the same error message, complained about getting a `404` response.

### What did you expect to happen?

It should give an error on the resource collision, but it should not state that it failed to run the `git fetch` because of a 404. [The file](https://raw.githubusercontent.com/argoproj/argo-cd/v2.7.8/notifications_catalog/install.yaml) I reference do exist.

### How can we reproduce it (as minimally and precisely as possible)?

```yaml
# kustomization.yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
- https://raw.githubusercontent.com/argoproj/argo-cd/v2.7.8/manifests/ha/install.yaml
- https://raw.githubusercontent.com/argoproj/argo-cd/v2.7.8/notifications_catalog/install.yaml
```

```console
$ kustomize build .
Error: accumulating resources: accumulation err='merging resources from 'https://raw.githubusercontent.com/argoproj/argo-cd/v2.7.8/notifications_catalog/install.yaml': may not add resource with an already registered id: ConfigMap.v1.[noGrp]/argocd-notifications-cm.[noNs]': failed to run '/usr/bin/git fetch --depth=1 https://raw.githubusercontent.com/argoproj/argo-cd HEAD': remote: 404: Not Found
fatal: repository 'https://raw.githubusercontent.com/argoproj/argo-cd/' not found
: exit status 128
```

### Expected output

Something along the lines of just this:

```
Error: accumulating resources: accumulation err='merging resources from 'https://raw.githubusercontent.com/argoproj/argo-cd/v2.7.8/notifications_catalog/install.yaml': may not add resource with an already registered id: ConfigMap.v1.[noGrp]/argocd-notifications-cm.[noNs]'
```

### Actual output

```
Error: accumulating resources: accumulation err='merging resources from 'https://raw.githubusercontent.com/argoproj/argo-cd/v2.7.8/notifications_catalog/install.yaml': may not add resource with an already registered id: ConfigMap.v1.[noGrp]/argocd-notifications-cm.[noNs]': failed to run '/usr/bin/git fetch --depth=1 https://raw.githubusercontent.com/argoproj/argo-cd HEAD': remote: 404: Not Found
fatal: repository 'https://raw.githubusercontent.com/argoproj/argo-cd/' not found
: exit status 128
```

### Kustomize version

v5.1.0

### Operating system

Linux

**Comment by annasong20** [comment]
Hi @lindhe, we agree that kustomize should no longer try to fetch the resource as a repo once it's retrieved valid yaml from the resource as an http file. 

The logic responsible for this behavior lives in the [`accumulateResources`](https://github.com/kubernetes-sigs/kustomize/blob/master/api/internal/target/kusttarget.go#L417) method under `api/internal/target`. This method has buggy logic resulting from its long history and patches that contributors have added as support for remote resources was introduced. 

Changes to `accumulateResources` is currently blocked by adequate test coverage in #4807 that I'm working on. We also plan on a significant refactoring of this code once #4807 is resolved to address issues like this, #4348, #4788, and #4935. @AryanSharma9917 we can't merge your fix for this issue until #4807 is resolved. When that happens, feel free to contribute to a larger refactoring!

**Comment by lindhe** [comment]
I think this is still on hold. Yeah?

**Comment by lindhe** [comment]
Not stale

**Comment by lindhe** [comment]
Not stale
